crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -5367,6 +5377,46 @@ impl<
     }
 }
 
+fn load_union_field_ptr_at_index_help<'a>(
+    layout_interner: &'a STLayoutInterner<'a>,
+    indices: &[u64],
+    layouts: &[InLayout<'a>],
+    mut offset: u32,
+) -> u32 {
+    debug_assert_ne!(indices.len(), 0);
+
+    let idx = indices[0];
+    for field in &layouts[..idx as usize] {
+        offset += layout_interner.stack_size(*field);
+    }
+
+    if indices.len() <= 1 {
+        return offset;
+    }
+    use LayoutRepr::*;
+    let enclosing_layout = layouts[idx as usize];
+    return match layout_interner.get_repr(enclosing_layout) {
+        Struct(layouts) => {
+            load_union_field_ptr_at_index_help(layout_interner, &indices[1..], layouts, offset)
```

Comment:
Doesn't need to happen now, but it would be nice if we could transform this into a loop instead of performing recursive calls. I think it would also simplify the logic. Something like


```
let mut indices = &indices[..];
let mut offset = 0;
while !indicies.is_empty() {
  let idx = indices[0];
  for field in &layouts[..idx as usize] {
      offset += layout_interner.stack_size(*field);
  }
  let enclosing_layout = layouts[idx as usize];
  match layout_interner.get_repr(enclosing_layout) {
    Struct(layouts) => { indices = &indices[1..]; }
    Union(UnionLayout::NonRecursive(tags)) => {
      // ...
    }
  }
}
offset
```